sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdOpenConnectionsHandler.java

```diff
@@ -22,7 +22,7 @@
 
 public class RntbdOpenConnectionsHandler implements IOpenConnectionsHandler {
     private static final Logger logger = LoggerFactory.getLogger(RntbdOpenConnectionsHandler.class);
-    private static int DEFAULT_CONNECTION_SEMAPHORE_TIMEOUT_IN_MINUTES = 10;
+    private static int DEFAULT_CONNECTION_SEMAPHORE_TIMEOUT_IN_MINUTES = 30;
```

Comment:
Should this setting be configurable in some way (env. variables etc)? 30 min is large value.

Reply:
It isn't harmful - after this timeout the semaphore acquisition would bail - so basically this is an upper bound for connection initialization. Restricting the upper bound isn't really necessary. If changing it again we would probably increase it even further - although I can't imagine right now that more than 30 minutes connection initialization would ever make sense.

Reply:
So, I don't think this needs to be configurable right now.